### ISSUE TITLE
A note about the remote timezone advice

### ISSUE THREAD

**Issue by jedwards1211** [moment/luxon#131]
> Don't make servers think about local times. Configure them to use UTC and write your server's code to work in UTC. Times can often be thought of as a simple count of epoch milliseconds; what you would call that time (e.g. 9:30) in what zone doesn't (again, often) matter.
Communicate times between systems in ISO 8601, like "2017-05-15T13:30:34Z" where possible (it doesn't matter if you use Z or some local offset; the point is that it precisely identifies the millisecond on the global timeline).
Where possible, only think of time zones as a formatting concern; your application ideally never knows that the time it's working with is called "9:00" until it's being rendered to the user.
Barring 3, do as much manipulation of the time (say, adding an hour to the current time) in the client code that's already running in the time zone where the results will matter.

I think you haven't personally had to work with business requirements that push the limits of the advice you're giving, so I want to tell you about the project I'm working on to broaden your view of the space of problems developers face.

The whole reason I switched to luxon was the need to determine when days start and end in remote timezones on both the client and server.  Our app schedules and displays production shifts at our clients' factories, which are in several different timezones.

It sounds like you're assuming a server will never have to know about a local timezone and can always work in UTC.  But many servers will need to determine if a given timestamp falls on a holiday in a remote timezone, as does ours.  Ours also needs to determine when days start and end in remote time zones for other parts of the scheduling process.

As far as the client, it sounds like you're assuming people using the website will only be dealing with dates for things located in the user's own timezone.  Our users need to be able to view data for any factory, even if it's in a different timezone than they're located in.  And we present all dates formatted for the remote timezone things are located in, even in date selectors.  So we can't do any of our manipulation of time (esp. finding starts and ends of days) in the client's local timezone.

For some problems, I think using pure semantic dates (month, day, year etc. instead of a numeric timestamp and communication in ISO8601) may be cleaner and easier to work with.  Sometimes I think it would be been better for our use case.  For example, the calendar and date picker components we're using unfortunately don't even support remote timezones so we have to wrap them with `keepCalendarTime: true` conversions.

**Comment by icambron** [comment]
From the docs:

>  But those aren't possible for some applications; you might need to work with times in zones other than the one the program is running in, for any number of reasons. And that's where Luxon's time zone support comes in.

Look, some of the advice doesn't apply to some applications. That's why I wrote all these capabilities in first place. Some of them, like the analog of `keepCalendarZone`, I even wrote into Moment ages ago to support my clients. Others reflect needs I had on enterprise products I built on other platforms. I promise I know that they're needed for real use cases. I would not have spent so much time writing them otherwise.

But most developers do not need them. I have helped support Moment for a long time, fielding hundreds of questions on related topics. Most people have simple needs and find all this done stuff confusing. Hence the advice.

It seems to me that what you're saying is just "this advice doesn't apply to me, so it is bad advice." Just because you have a good reason to let go of the handlebars does not make "hold the handlebars" bad advice.

See also #81
